### DISCUSSION TITLE
step by step guide for installtion and integration

### DISCUSSION THREAD

**Discussion by abhishek9699** [custompbx/custompbx#11] [General]
Hi All,

I have multipe insatce of FS running in my Org. Need to step by guide to Custom PBX installation and integration.

**Comment by zusrut** [comment]
Hi. First of all checkout project [wiki](https://github.com/custompbx/custompbx/wiki/install.md) .
You need at leas one postgres database and one custompbx service per freeswitch instance.

To use same database for different instanses you need to set instance_id in config.json file.

**Reply by abhishek9699** [reply]
@zusrut, Thanks for the reply. Could you please suggest how we can have instance_id in config.json. 

Though I have multiple FS running, all are having same configuration.

**Reply by zusrut** [reply]
Ah yes **instance_id** is **switchname** in conf.json - it need to be different per instatce. if empty - will be equal to hostname

BTW all instances still be autonomuos and separate with custompbx(at least in the current version)

**Reply by stachura93** [reply]
@zusrut 
I’m wondering how to scale CustomPBX. At first, I was thinking about running several FreeSWITCH instances and came across this thread.

My initial idea was to run multiple instances of both CustomPBX and FreeSWITCH, connect them to Postgres with some replication and database partitioning.

I’m also considering using Kamailio to route traffic to the right FreeSWITCH and possibly separate domains at the CustomPBX level.

Would you be able to suggest a good direction to take?

**Reply by zusrut** [reply]
Hello @stachura93, custompbx was designed to work with one freeswitch instance. But both FS and CPBX can share same databese between different instances by **instance_id** as i mentioned above.
